docs/java-rest/high-level/apis/search.asciidoc

```diff
@@ -1,4 +1,103 @@
 [[java-rest-high-search]]
 === Search API
 
-To be documented.
+[[java-rest-high-document-search-request]]
+==== Search Request
+
+The `SearchRequest` is used for any operation that has to do with retrieving 
+documents, aggregations, suggestions and also offers ways of requesting 
+highlighting on the resulting documents.
+
+In its most basic form, a query can be added to the request like this: 
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-basic]
+--------------------------------------------------
+
+<1> Creates the `SeachRequest`. Without arguments this runs against all indices.
+<2> Most details about the search can be added to the `SearchSourceBuilder`.
+<3> Add a `match_all` query to the `SearchSourceBuilder`.
+
+When the search is executed, it returns a `SearchResponse` object that serves
```

Comment:
I would prefer to follow other APIs docs and first talk about the request and all its options, then have a section about the response and what to do with it